### ISSUE TITLE
Display non-empty fields

### ISSUE THREAD

**Issue by cneidle** [DCS-LCSR/SignStream3#686]
On opening an utterance, by default the non-empty fields (and only those) should be displayed.

So, I click on the non-empty fields button, and that is fine.

But when I delete one item, the field display reverts back to showing all sorts of empty fields.

<img width="1123" alt="Screen Shot 2023-08-31 at 4 15 57 PM" src="https://github.com/DCS-LCSR/SignStream3/assets/13629122/a713aaab-9035-44b3-a20a-19d5a3655785">

When I delete the item highlighted above in red, this happens:

<img width="1461" alt="Screen Shot 2023-08-31 at 4 16 11 PM" src="https://github.com/DCS-LCSR/SignStream3/assets/13629122/fefdf701-1883-49da-97cd-1be8eed44cb5">

When I click on the non-empty fields button, it returns to the proper display

<img width="1478" alt="Screen Shot 2023-08-31 at 4 16 19 PM" src="https://github.com/DCS-LCSR/SignStream3/assets/13629122/fdc3cb1a-e79b-47e0-9c5b-36c34ceca3ea">

**Comment by cneidle** [comment]
The above is using SignStream 3.4.1 on Mac OS 11.7.9.  The file was DSP_Sentences-DL.

**Comment by cneidle** [comment]
This is a serious issue, but it is a little bit complicated; we should discuss.

**Comment by gregorydimitriadis** [comment]
Main issue at hand for this: do not trigger the default view on a delete event. This is what's occurring now and resulting in the inconsistency.

**Comment by gregorydimitriadis** [comment]
Carol, to clarify this portion: "by default the non-empty fields (and only those) should be displayed."

If we were to do that, would that apply to nonmanual/gloss/gesture alike? Or just nonmanuals specifically.

**Comment by cneidle** [comment]
I would think that our current display of basic gloss/gesture fields
should remain as it is now.

For optionally specified information about gestures (e.g., location,
function, etc.), then it should be like the non-manuals, in that 
the default display should be non-empty, with the option of 
clicking on the button to display those that are selected for display.

Thanks!
  Carol

**Comment by cneidle** [comment]
Non-empty fields are not currently displayed by default.  I open this utterance, for example:

<img width="1532" alt="Screen Shot 2023-11-30 at 11 30 45 AM" src="https://github.com/DCS-LCSR/SignStream3/assets/13629122/5bf435d3-4080-4b59-a7d2-c5b339daf2b1">

The hp:jut and hm:nod fields are displayed although they are empty.

The display non empty button does work properly, however.  And editing fields does not affect the display non empty selection...  So that much is good!

**Comment by cneidle** [comment]
This remains a serious problem (in v. 3.5.0).  If I open a file, in order to display non-empty fields, I need to click that button for each sentence.  If I save and re-open the file, my selection of Display non-empty has been lost, and I need to do this again for every sentence.  There ought to be some global option to display non-empty for the whole file. @gregorydimitriadis
